presentation/src/main/java/com/mashup/gabbangzip/sharedalbum/presentation/ui/groupcreation/name/GroupCreationNameScreen.kt

```diff
@@ -0,0 +1,92 @@
+package com.mashup.gabbangzip.sharedalbum.presentation.ui.groupcreation.name
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.imePadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.mashup.gabbangzip.sharedalbum.presentation.R
+import com.mashup.gabbangzip.sharedalbum.presentation.theme.Gray0
+import com.mashup.gabbangzip.sharedalbum.presentation.theme.Gray80
+import com.mashup.gabbangzip.sharedalbum.presentation.theme.PicTypography
+import com.mashup.gabbangzip.sharedalbum.presentation.theme.SharedAlbumTheme
+import com.mashup.gabbangzip.sharedalbum.presentation.ui.common.PicButton
+import com.mashup.gabbangzip.sharedalbum.presentation.ui.common.PicProgressBar
+import com.mashup.gabbangzip.sharedalbum.presentation.ui.common.PicTextField
+import com.mashup.gabbangzip.sharedalbum.presentation.utils.hideKeyboardOnOutsideClicked
+
+@Composable
+fun GroupCreationNameScreen(
+    onBackButtonClicked: () -> Unit,
+    onNextButtonClicked: () -> Unit,
+) {
+    var name by remember { mutableStateOf("") }
+
```

Comment:
```suggestion
    var name by remember { mutableStateOf("") }
    val buttonEnabled by remember { derivedStateOf { name.isNotBlank() } }
```
derivedStateOf 를 통해 buttonEnabled 를 위한 새로운 State 를 만들어보는 것은 어떨까?

그리고

```kotlin
        PicButton(
            modifier = Modifier
                .fillMaxWidth()
                .imePadding()
                .padding(bottom = 16.dp),
            text = stringResource(id = R.string.group_creation_button_next),
            isRippleClickable = true,
            enable = buttonEnabled,
            onButtonClicked = onNextButtonClicked,
        )
```
이런식으로 적용하자는 아이디어인데..

https://velog.io/@beokbeok/%EC%96%B8%EC%A0%9C-derivedStateOf%EB%A5%BC-%EC%8D%A8%EC%95%BC%ED%95%A0%EA%B9%8C
이전에 위 글을 읽었던 기억이 있어서 이런 상황에서 딱 derivedStateOf 를 사용하기 좋을 것 같아서 제안해봐!

Reply:
오옹 아니면 아예 name이랑 enabled랑 뷰모델에 두는건 어뗘?

Reply:
> 오옹 아니면 아예 name이랑 enabled랑 뷰모델에 두는건 어뗘?

이게.. name 이 Ui를 나타내는 State 의 일부이긴 한데... name 을 ViewModel 쪽으로 빼주기엔 EditText 가 변경될 때 마다 viewModel update 가 호출되고, 이에 맞게 기존 UiState 를 copy() 하고 name 값 수정해서 flow 로 내려주는 작업이 추가될 것 같은데 이러한 작업까지는 불필요하지 않을까?! 
뭔가.. Composable 의 Stateless 구조를 위해 ViewModel 이 안받아도 되는 작업을 떠안아버린 느낌인 것 같아서...!!